experimental/streaming/exporter/reader/format/format.go

```diff
@@ -61,53 +65,45 @@ func AppendEvent(buf *strings.Builder, data reader.Event) {
 			buf.WriteString(" >")
 		}
 
-	case reader.FINISH_SPAN:
+	case observer.FINISH_SPAN:
 		buf.WriteString("finish ")
 		buf.WriteString(data.Name)
 
 		buf.WriteString(" (")
 		buf.WriteString(data.Duration.String())
 		buf.WriteString(")")
 
-	case reader.ADD_EVENT:
+	case observer.ADD_EVENT:
 		buf.WriteString("event: ")
 		buf.WriteString(data.Message)
 		buf.WriteString(" (")
 		data.Attributes.Foreach(func(kv core.KeyValue) bool {
-			buf.WriteString(" " + kv.Key.Variable.Name + "=" + kv.Value.Emit())
+			buf.WriteString(" ")
+			buf.WriteString(kv.Key.Variable.Name)
+			buf.WriteString("=")
+			buf.WriteString(kv.Value.Emit())
 			return true
 		})
 		buf.WriteString(")")
 
-	case reader.MODIFY_ATTR:
-		buf.WriteString("modify attr")
-	case reader.RECORD_STATS:
-		buf.WriteString("record")
-
-		for _, s := range data.Stats {
-			f(false)(core.Key{
-				Variable: s.Measure.V(),
-			}.Float64(s.Value))
-
-			buf.WriteString(" {")
-			i := 0
-			s.Tags.Foreach(func(kv core.KeyValue) bool {
-				if i != 0 {
-					buf.WriteString(",")
-				}
-				i++
-				buf.WriteString(kv.Key.Variable.Name)
-				buf.WriteString("=")
-				buf.WriteString(kv.Value.Emit())
-				return true
-			})
-			buf.WriteString("}")
+	case observer.MODIFY_ATTR:
+		buf.WriteString(data.Type.String())
+
+	case observer.UPDATE_METRIC:
+		formatMetricUpdate(buf, data.Type, data.Measurement)
+
+	case observer.BATCH_METRIC:
+		buf.WriteString("BATCH ")
+		for _, m := range data.Measurements {
+			formatMetricUpdate(buf, observer.UPDATE_METRIC, m)
```

Comment:
Maybe would be better if each metric was bracketed, so we can tell one from another?